api/profilerecording/v1alpha1/profilerecording_types.go

```diff
@@ -50,6 +67,13 @@ type ProfileRecordingSpec struct {
 	// +kubebuilder:validation:Enum=bpf;logs
 	Recorder ProfileRecorder `json:"recorder"`
 
+	// Whether or how to merge recorded profiles. Can be one of "none" or "containers".
```

Comment:
Ok - this answers a question I had elsewhere.

What other types of merge strategies do we anticipate? If I use the `containers` strategy, my profile recording will be an aggregate of everything that happened across the containers. Do we anticipate merging recordings across other entities , like Deployments and ReplicaSets?

For example, would I be able to distill recordings from multiple Deployments if I set `deployment` as the merge strategy (assuming we implemented something like that?)

To merge, or not to merge, feels like a binary decision and the setting here eludes to potential functionality that adds another dimension to record merging, so I'm trying to understand that bit.

Reply:
I was thinking that maybe the workload level might be a good candidate for future or even a profile recording level (so, everything recorded using a single profileRecording would result in a single policy), but the container-level merging seemed like a good start in terms of being less granular than per-container-instance which is the `none` default but still not merging too much. As I said elsewhere, the profile merging is a bit of a security/usability tradeoff where you trade not having to deal with potentially dozens of policies after a recording session vs. the danger that one of the policies record something you don't want in the resulting policy.

I wonder if @saschagrunert or @JAORMX have other opinions?

Reply:
I agree leaving the option for further higher level merges is a good path forward.